OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -220,25 +221,23 @@ public SlotDropDownOption(string title, string order, Func<bool> selected)
 		}
 
 		public static void ShowColorDropDown(DropDownButtonWidget color, Session.Client client,
-			OrderManager orderManager, World world, ColorPreviewManagerWidget preview)
+			OrderManager orderManager, WorldRenderer worldRenderer, ColorPickerManagerInfo colorManager)
 		{
 			Action onExit = () =>
 			{
-				if (client.Bot == null)
+				if (client == orderManager.LocalClient)
 				{
-					Game.Settings.Player.Color = preview.Color;
+					Game.Settings.Player.Color = colorManager.Color;
 					Game.Settings.Save();
 				}
 
 				color.RemovePanel();
-				orderManager.IssueOrder(Order.Command($"color {client.Index} {preview.Color}"));
+				orderManager.IssueOrder(Order.Command($"color {client.Index} {colorManager.Color}"));
 			};
 
-			Action<Color> onChange = c => preview.Color = c;
-
-			var colorChooser = Game.LoadWidget(world, "COLOR_CHOOSER", null, new WidgetArgs()
+			var colorChooser = Game.LoadWidget(worldRenderer.World, "COLOR_CHOOSER", null, new WidgetArgs()
```

Comment:
cant we just use orderManager.World here?

Reply:
`orderManager.World` refers to the world that hasn't yet been created for the game (null at this point), `worldRenderer.World` refers to the shellmap. This is all a mess that would be nice to clean up one day, but for now we can't use `orderManager.World`.